hpx/runtime/parcelset/parcel.hpp

```diff
@@ -149,182 +104,109 @@ namespace hpx { namespace parcelset
-          , typename...Args
-          , typename Enable =
-                typename std::enable_if<traits::is_action<Action>::value>::type
-          >
-        parcel(
-            std::vector<naming::id_type> const& dests,
-            std::vector<naming::address> const& addrs,
-            Action,
-            Args &&... args
-        )
-          : dests_(dests),
-            addrs_(addrs),
-            action_(new actions::transfer_action<Action>(std::forward<Args>(args)...))
-        {
-            HPX_ASSERT(is_valid());
-        }
-#endif
-
-        template <
-            typename Continuation
-          , typename Action
-          , typename...Args
-          , typename Enable
-                = typename std::enable_if<
-                    traits::is_action<Action>::value
-                 && traits::is_continuation<Continuation>::value
-                 && !std::is_same<
-                        Continuation
-                      , std::unique_ptr<actions::continuation>
-                    >::value
-                >::type
-          >
-        parcel(
-            naming::id_type const& dests,
-            naming::address const& addrs,
-            Continuation && cont,
-            Action,
-            Args &&... args
-        )
-#if defined(HPX_SUPPORT_MULTIPLE_PARCEL_DESTINATIONS)
-          : dests_(1, dests),
-            addrs_(1, addrs),
-#else
-          : dests_(dests),
-            addrs_(addrs),
-#endif
-            cont_(
-                new typename util::decay<Continuation>
-                    ::type(std::forward<Continuation>(cont))
-            ),
-            action_(new actions::transfer_action<Action>(std::forward<Args>(args)...))
-        {
-            HPX_ASSERT(is_valid());
-        }
-
-        template <
-            typename Action
-          , typename...Args
-          , typename Enable
-                = typename std::enable_if<
-                    traits::is_action<Action>::value
-                >::type
-          >
+    private:
         parcel(
-            naming::id_type const& dests,
-            naming::address const& addrs,
+            naming::gid_type&& dest,
+            naming::address&& addr,
             std::unique_ptr<actions::continuation> cont,
-            Action,
-            Args &&... args
+            std::unique_ptr<actions::base_action> act,
+            bool action_args_bitwise = false
         )
-#if defined(HPX_SUPPORT_MULTIPLE_PARCEL_DESTINATIONS)
-          : dests_(1, dests),
-            addrs_(1, addrs),
-#else
-          : dests_(dests),
-            addrs_(addrs),
-#endif
+          : data_(std::move(dest), std::move(addr)),
             cont_(std::move(cont)),
-            action_(new actions::transfer_action<Action>(std::forward<Args>(args)...))
+            action_(std::move(act)),
+            size_(0),
+            action_args_bitwise_(action_args_bitwise)
         {
-            HPX_ASSERT(is_valid());
+//             HPX_ASSERT(is_valid());
```

Comment:
Am 02.09.2016 5:38 nachm. schrieb "Hartmut Kaiser" <notifications@github.com
>:
>
> In hpx/runtime/parcelset/parcel.hpp:
>
> > { > - HPX_ASSERT(is_valid()); > +// HPX_ASSERT(is_valid());
>
> Please re-enable this assert.

No, it will fire now :p
With the changes, a constructed parcel doesn't necessarily need to be
valid. The target might be invalid and set later on.

>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.